docs/sdk-regeneration-pipeline.md

```diff
@@ -0,0 +1,137 @@
+# SDK Regeneration Pipeline Guideline
+
+This document provides a guideline about the SDK regeneration pipeline for TypeSpec Go emitter developers.
+
+## Pipeline Overview
+
+> **Pipeline Link**: [SDK Regeneration Pipeline](https://dev.azure.com/azure-sdk/internal/_build?definitionId=7860)
+
+The SDK regeneration pipeline regenerates Azure SDK for Go packages using any branch of TypeSpec Go emitter with the same API version as the released SDK (which to make sure SDK changes are due to TypeSpec or Go emitter release, not API version). This pipeline validates emitter changes and ensures that SDK packages stay up-to-date with TypeSpec changes and emitter changes.
```

Comment:
Not quite understand this,  'using any branch of TypeSpec Go emitter',  could you explain with more details?